aaerec/aae.py

```diff
@@ -220,11 +220,36 @@ def zero_grad(self):
         self.enc.zero_grad()
         self.dec.zero_grad()
 
-    def ae_step(self, batch, condition=None):
-        """ Perform one autoencoder training step """
+    # why is this double? to AdversarialAutoEncoder
+    def ae_step(self, batch, condition=None, conditions=[]):
+
+        """
+        Perform one autoencoder training step
+            :param batch: # self.enc() based in models
+            :param condition: probably some matrix (.shape works)
+            :return: I belive: binary_cross_entropy for this step
+            """
+
+        # why is this double to AdversarialAutoEncoder? Lukas: it's likely the two models will diverge
+        # what is relationship to train in DecodingRecommender? Train only uses Condition. Those are implementet seperately
+
         z_sample = self.enc(batch)
-        if condition is not None:
-            z_sample = torch.cat((z_sample, condition), 1)
+
+        #if condition is not None:
```

Comment:
Why is this commented out here, but not in the other case?

Reply:
Make the two AE steps the same.

Reply:
It is commented out here, because I used this instance for testing, but didn't want to change everything back and forth before the tests were finished. As soon as the ae_step works for the AutoEncoder class, it will be copied to the AdversarialAutoEncoder.